### PR TITLE
complete s2s e2e test

### PR DESCRIPTION
**Pull request by woutslakhorst**

### CODE REVIEW

File: auth/api/iam/api.go [hunk 1]
```diff
@@ -170,18 +170,21 @@ func (r Wrapper) IntrospectAccessToken(ctx context.Context, request IntrospectAc
...
 		// Return 200 + 'Active = false' when token is invalid or malformed
+		log.Logger().Debug("IntrospectAccessToken: missing token")
 		return IntrospectAccessToken200JSONResponse{}, nil
 	}
 
 	token := AccessToken{}
 	if err := r.accessTokenStore().Get(request.Body.Token, &token); err != nil {
 		// Return 200 + 'Active = false' when token is invalid or malformed
+		log.Logger().Debug("IntrospectAccessToken: failed to get token from store")
 		return IntrospectAccessToken200JSONResponse{}, err
```
Comment (reinkrul) [inline auth/api/iam/api.go:181]: error is still logged, right?

File: e2e-tests/oauth-flow/rfc021/node-A/nginx.conf [hunk 2]
```diff
@@ -43,5 +44,18 @@ http {
...
         proxy_pass http://nodeA-backend;
       }
+
+      location /resource {
+          js_content oauth2.introspectAccessToken;
+      }
+
+      # Location in javascript subrequest.
+      # this is needed to set headers and method
+	  location /_oauth2_send_request {
```
Comment (reinkrul) [inline e2e-tests/oauth-flow/rfc021/node-A/nginx.conf:54]: A year from now on you'll read this and think; who came up with this...

File: e2e-tests/oauth-flow/rfc021/node-A/nginx.conf [hunk 3]
```diff
@@ -1,18 +1,19 @@
+load_module /usr/lib/nginx/modules/ngx_http_js_module.so;
```
Comment (reinkrul) [inline e2e-tests/oauth-flow/rfc021/node-A/nginx.conf:1]: Ah, it's not actually JavaScript; https://nginx.org/en/docs/njs/index.html

Or it is, but a custom implementation..?

File: e2e-tests/oauth-flow/rfc021/node-A/nuts.yaml [hunk 4]
```diff
@@ -6,6 +6,7 @@ datadir: /opt/nuts/data
 http:
   default:
     address: :1323
+    log: metadata-and-body
```
Comment (reinkrul) [inline e2e-tests/oauth-flow/rfc021/node-A/nuts.yaml:9]: still needed?

File: e2e-tests/oauth-flow/rfc021/node-A/oauth2.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+function introspectAccessToken(r) {
+    // strip the first 8 chars
+    var token = "token=" + r.headersIn['Authorization'].substring(7);
```
Comment (reinkrul) [inline e2e-tests/oauth-flow/rfc021/node-A/oauth2.js:3]: does this always have the right encoding?

File: e2e-tests/oauth-flow/rfc021/run-test.sh [hunk 6]
```diff
@@ -66,14 +66,14 @@ fi
...
 echo "------------------------------------"
-#RESPONSE=$(docker compose exec nodeB curl --insecure --cert /etc/nginx/ssl/server.pem --key /etc/nginx/ssl/key.pem https://nodeA:443/ping -H "Authorization: bearer $(cat ./node-B/data/accesstoken.txt)" -v)
-#if echo $RESPONSE | grep -q "pong"; then
-#  echo "success!"
-#else
-#  echo "FAILED: Could not ping node-A" 1>&2
-#  echo $RESPONSE
-#  exitWithDockerLogs 1
-#fi
+RESPONSE=$(docker compose exec nodeB curl --http1.1 --insecure --cert /etc/nginx/ssl/server.pem --key /etc/nginx/ssl/key.pem https://nodeA:443/resource -H "Authorization: bearer $(cat ./node-B/data/accesstoken.txt)" -v)
```
Comment (reinkrul) [inline e2e-tests/oauth-flow/rfc021/run-test.sh:69]: should we also have a not-OK test?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: auth/api/iam/api.go (hunk 1 above)
Comment (woutslakhorst) [reply auth/api/iam/api.go:181]: Nothing was logged, now there's some debug log. We may want to create an issue to 'fix' the logging for introspection calls?

File: e2e-tests/oauth-flow/rfc021/node-A/nginx.conf (hunk 2 above)
Comment (woutslakhorst) [reply e2e-tests/oauth-flow/rfc021/node-A/nginx.conf:54]: I'll add my name.... I mean the link to the blog post.

File: e2e-tests/oauth-flow/rfc021/node-A/nuts.yaml (hunk 4 above)
Comment (woutslakhorst) [reply e2e-tests/oauth-flow/rfc021/node-A/nuts.yaml:9]: no

File: e2e-tests/oauth-flow/rfc021/node-A/oauth2.js (hunk 5 above)
Comment (woutslakhorst) [reply e2e-tests/oauth-flow/rfc021/node-A/oauth2.js:3]: we use 128bit b64 encoded strings, so for us: yes.

File: e2e-tests/oauth-flow/rfc021/run-test.sh (hunk 6 above)
Comment (woutslakhorst) [reply e2e-tests/oauth-flow/rfc021/run-test.sh:69]: I'd like to have e2e tests to make sure the integration isn't broken at some point. Various error conditions should be tested by unit-tests. At this point a no-OK test would largely test the nginx config.

File: e2e-tests/oauth-flow/rfc021/node-A/oauth2.js (hunk 5 above)
Comment (reinkrul) [reply e2e-tests/oauth-flow/rfc021/node-A/oauth2.js:4]: I raw thinking of standard/raw/URL base64 encoding, but if it works...

**Review by reinkrul — APPROVED** [review]

**Pull request merged**